internal/experiment/generator.go

```diff
@@ -0,0 +1,152 @@
+	// This bridges the gap between the powerful selection logic that is implemented vs the simple
+	// selection configuration that is actually exposed on the Application.
+
+	// Always add container resource selectors, conditionally with an explicit label selector
+	var crsLabelSelector string
+	if g.Application.Parameters != nil && g.Application.Parameters.ContainerResources != nil {
+		crsLabelSelector = g.Application.Parameters.ContainerResources.LabelSelector
+	}
+	g.ContainerResourcesSelectors = []generation.ContainerResourcesSelector{
+		{
+			GenericSelector: scan.GenericSelector{
+				Group:         "apps|extensions",
+				Kind:          "Deployment",
+				LabelSelector: crsLabelSelector,
+			},
+			Path:               "/spec/template/spec/containers",
+			CreateIfNotPresent: true,
+		},
+		{
+			GenericSelector: scan.GenericSelector{
+				Group:         "apps|extensions",
+				Kind:          "StatefulSet",
+				LabelSelector: crsLabelSelector,
+			},
+			Path:               "/spec/template/spec/containers",
+			CreateIfNotPresent: true,
+		},
+	}
+
+	// Only add replica selectors if the parameter is explicitly configured
+	if g.Application.Parameters != nil && g.Application.Parameters.Replicas != nil {
+		g.ReplicaSelectors = []generation.ReplicaSelector{
+			{
+				GenericSelector: scan.GenericSelector{
+					Group:         "apps|extensions",
+					Kind:          "Deployment",
+					LabelSelector: g.Application.Parameters.Replicas.LabelSelector,
+				},
+				Path:               "/spec/replicas",
+				CreateIfNotPresent: true,
+			},
+			{
+				GenericSelector: scan.GenericSelector{
+					Group:         "apps|extensions",
+					Kind:          "StatefulSet",
+					LabelSelector: g.Application.Parameters.Replicas.LabelSelector,
+				},
+				Path:               "/spec/replicas",
+				CreateIfNotPresent: true,
+			},
+		}
+	}
+}
+
+// Execute the experiment generation pipeline, sending the results to the supplied writer.
+func (g *Generator) Execute(output kio.Writer) error {
+	return kio.Pipeline{
+		Inputs: []kio.Reader{
+			g.Application.Resources,
+		},
+		Filters: []kio.Filter{
+			&konjure.Filter{
+				Depth:         100,
+				DefaultReader: g.DefaultReader,
+				KeepStatus:    true,
+			},
+			g.newScannerFilter(),
+			&generation.ApplicationFilter{Application: &g.Application},
+			kio.FilterAll(yaml.Clear("status")),
+			&filters.FormatFilter{UseSchema: true},
+		},
+		Outputs: []kio.Writer{
+			output,
+		},
+	}.Execute()
+}
+
+func (g *Generator) newScannerFilter() kio.Filter {
+	scanner := &scan.Scanner{
+		Transformer: &generation.Transformer{
+			DefaultExperimentName:       application.ExperimentName(&g.Application),
+			MergeGenerated:              len(g.Application.Scenarios) > 1,
+			IncludeApplicationResources: g.IncludeApplicationResources,
+		},
+	}
+
+	for i := range g.ContainerResourcesSelectors {
+		scanner.Selectors = append(scanner.Selectors, &g.ContainerResourcesSelectors[i])
+	}
```

Comment:
should we skip the range and do 
```
scanner.Selectors = append(scanner.Selectors, &g.ContainerResourcesSelectors...)
```

Reply:
No, because the elements need to have their type converted independently.

Reply:
Doesn't that get handled by the variadic parameter?

Reply:
err, I think I got what you're saying.

Reply:
https://golang.org/doc/faq#convert_slice_of_interface